src/main/java/com/toopher/ToopherAPIDemo.java

```diff
@@ -1,149 +1,165 @@
-        	api = new ToopherAPI(env.get("TOOPHER_CONSUMER_KEY"), env.get("TOOPHER_CONSUMER_SECRET"), base_uri);
+
+        ToopherApi api;
+        if (env.containsKey("TOOPHER_CONSUMER_KEY") && env.containsKey("TOOPHER_CONSUMER_SECRET")) {
+            api = new ToopherApi(env.get("TOOPHER_CONSUMER_KEY"), env.get("TOOPHER_CONSUMER_SECRET"), base_uri);
         } else {
-	        System.out.println("");
-	        System.out.println("Setup Credentials");
-	        System.out.println("--------------------------------------");
-	        System.out.println("Enter your requester credentials (from https://dev.toopher.com)");
-	        System.out.print("Consumer key: ");
-	        String consumerKey = in.nextLine();
-	        System.out.print("Consumer secret: ");
-	        String consumerSecret = in.nextLine();
-	
-			api = new ToopherAPI (consumerKey, consumerSecret, base_uri);
+            System.out.println("");
+            System.out.println("Setup Credentials");
+            System.out.println("--------------------------------------");
+            System.out.println("Enter your requester credential details (from https://dev.toopher.com)");
+            System.out.print("Toopher Consumer Key: ");
+            String consumerKey = in.nextLine();
+            System.out.print("Toopher Consumer Secret: ");
+            String consumerSecret = in.nextLine();
+
+            api = new ToopherApi(consumerKey, consumerSecret, base_uri);
         }
-        
-		String pairingId;
-		while (true) {
-			String pairingPhrase;
-			while (true) {
-				System.out.println("Step 1: Pair requester with phone");
-				System.out.println("--------------------------------------");
-				System.out.println("Pairing phrases are generated on the mobile app");
-				System.out.print("Enter pairing phrase: ");
-				pairingPhrase = in.nextLine();
-
-				if (pairingPhrase.length() == 0) {
-					System.out.println("Please enter a pairing phrase to continue");
-				} else {
-					break;
-				}
-			}
-
-			System.out.print(String.format("Enter a username for this pairing [%s]: ", DEFAULT_USERNAME));
-			String userName = in.nextLine();
-			if (userName.length() == 0) {
-				userName = DEFAULT_USERNAME;
-			}
-
-			System.out.println("Sending pairing request...");
-
-			try {
-				PairingStatus pairingStatus = api.pair(pairingPhrase, userName);
-				pairingId = pairingStatus.id;
-				break;
-			} catch (RequestError err) {
-				System.out.println(String.format("The pairing phrase was not accepted (reason:%s)", err.getMessage()));
-			}
-		}
-
-		while (true) {
-			System.out.println("Authorize pairing on phone and then press return to continue.");
-			in.nextLine();
-			System.out.println("Checking status of pairing request...");
-
-			try {
-				PairingStatus pairingStatus = api.getPairingStatus(pairingId);
-				if (pairingStatus.enabled) {
-					System.out.println("Pairing complete");
+
+        String pairingId;
```

Comment:
If we're pushing developers toward authenticating by username (née Zero Requester Storage), shouldn't we update the demo to use that as well?